### DISCUSSION TITLE
Databricks or SparkSQL support?

### DISCUSSION THREAD

**Discussion by boblannon** [theory/sqitch#834] [Q&A]
I really love sqitch's approach to change management and I'd like to try maintaining the catalogs, databases and schemas in my  Databricks platform using it.  Has anyone tried doing this? Is it possible to do it using any of the existing integrations + the databricks ODBC driver?

**Comment by theory** [comment]
I've not used Databricks, but I would assume it would require implementing a new engine to complement the [existing engines](https://github.com/sqitchers/sqitch/tree/develop/lib/App/Sqitch/Engine).

**Reply by pansharma11** [reply]
@theory regarding the new engine implementation for databricks, do you know if there is any planned roadmap for it's implementation ? or how it works if we need this functionality, should we research, implement & contribute ? can you please guide me here.

**Reply by theory** [reply]
I have no plans to implement it, given that I've not used it. Your contribution would be most welcome, however, and I'd be happy to help get it integrated. Somewhere I wrote some guidelines of how to add a new engine; let see where I hid them…

Bah, not finding them. I think the best place to start is looking at the last time a new engine was added. The [Snowflake PR](https://github.com/sqitchers/sqitch/pull/397) is probably the best guide. I'd add two quick notes here, though:

1. You'll need to add a new URI to [URI::db](https://github.com/libwww-perl/URI-db). It's pretty simple; here is the [PR to add URI::clickhouse](https://github.com/libwww-perl/URI-db/pull/18).
2. If Databricks is similar to one of the existing engines, you might be able to subclass that engine to save effort. The [Cockroach PR](https://github.com/sqitchers/sqitch/pull/643) took that approach.
